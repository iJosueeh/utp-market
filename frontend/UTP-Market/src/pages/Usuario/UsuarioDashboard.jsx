import React, { useState } from 'react';
import Sidebar from './Sidebar';
import DatosPersonales from './DatosPersonales';
import TusResenas from './TusReseÃ±as';
import TusProductos from './TusProductos';
import Favoritos from './Favoritos';
import HistorialDeCompra from './HistorialDeCompra';
import EstadisticasPersonales from './EstadisticasPersonales';
import NotFoundSection from './NotFoundSection';
import { Routes, Route } from 'react-router-dom';

const UsuarioDashboard = () => {
  const [isSidebarOpen, setIsSidebarOpen] = useState(false);

  const toggleSidebar = () => {
    setIsSidebarOpen(!isSidebarOpen);
  };

  return (
    <div className="usuario-dashboard">
      <Sidebar isOpen={isSidebarOpen} toggle={toggleSidebar} />
      <div className="main-content">
        <button className={`sidebar-toggle d-lg-none ${isSidebarOpen ? 'hidden' : ''}`} onClick={toggleSidebar} data-testid="sidebar-toggle">
          <i className="bi bi-list"></i>
        </button>
        <div className="content-wrapper">
          <Routes>
            <Route index element={<DatosPersonales />} />
            <Route path="datos-personales" element={<DatosPersonales />} />
            <Route path="tus-resenas" element={<TusResenas />} />
            <Route path="tus-productos" element={<TusProductos />} />
            <Route path="favoritos" element={<Favoritos />} />
            <Route path="historial-de-compra" element={<HistorialDeCompra />} />
            <Route path="estadisticas-personales" element={<EstadisticasPersonales />} />
            <Route path="*" element={<NotFoundSection />} />
          </Routes>
        </div>
      </div>
      <style jsx>{`
        .usuario-dashboard {
          display: flex;
        }
        .main-content {
          flex: 1;
          padding: 20px;
          position: relative;
        }
        .sidebar-toggle {
          position: absolute;
          top: 20px;
          left: 20px;
          z-index: 1001;
          background: #222222;
          color: white;
          border: none;
          border-radius: 50%;
          width: 40px;
          height: 40px;
          font-size: 20px;
          display: flex;
          align-items: center;
          justify-content: center;
          padding: 0;
          line-height: 0;
        }
        .sidebar-toggle.hidden {
          display: none;
        }
        @media (max-width: 992px) {
          .content-wrapper {
            padding-top: 50px;
          }
        }
      `}</style>
    </div>
  );
};

export default UsuarioDashboard;
