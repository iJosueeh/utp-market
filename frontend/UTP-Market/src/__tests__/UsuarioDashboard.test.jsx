import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import { describe, test, expect, vi } from 'vitest';
import UsuarioDashboard from '../pages/Usuario/UsuarioDashboard';

// Mock the components to isolate the test
vi.mock('../pages/Usuario/SideBar', () => ({ default: () => <div>Sidebar</div> }));
vi.mock('../pages/Usuario/DatosPersonales', () => ({ default: () => <div>Datos Personales</div> }));
vi.mock('../pages/Usuario/TusReseñas', () => ({ default: () => <div>Tus Reseñas</div> }));
vi.mock('../pages/Usuario/TusProductos', () => ({ default: () => <div>Tus Productos</div> }));
vi.mock('../pages/Usuario/Favoritos', () => ({ default: () => <div>Favoritos</div> }));
vi.mock('../pages/Usuario/HistorialDeCompra', () => ({ default: () => <div>Historial De Compra</div> }));
vi.mock('../pages/Usuario/EstadisticasPersonales', () => ({ default: () => <div>Estadisticas Personales</div> }));

describe('UsuarioDashboard', () => {
  const renderWithRouter = (initialEntries) => {
    return render(
      <MemoryRouter initialEntries={initialEntries}>
        <Routes>
          <Route path="/usuario/*" element={<UsuarioDashboard />} />
        </Routes>
      </MemoryRouter>
    );
  };

  test('renders Sidebar and default route (DatosPersonales)', () => {
    const { asFragment } = renderWithRouter(['/usuario']);
    expect(screen.getByText('Sidebar')).toBeInTheDocument();
    expect(screen.getByText('Datos Personales')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders TusResenas component on /tus-resenas route', () => {
    renderWithRouter(['/usuario/tus-resenas']);
    expect(screen.getByText('Tus Reseñas')).toBeInTheDocument();
  });

  test('renders TusProductos component on /tus-productos route', () => {
    renderWithRouter(['/usuario/tus-productos']);
    expect(screen.getByText('Tus Productos')).toBeInTheDocument();
  });

  test('renders Favoritos component on /favoritos route', () => {
    renderWithRouter(['/usuario/favoritos']);
    expect(screen.getByText('Favoritos')).toBeInTheDocument();
  });

  test('renders HistorialDeCompra component on /historial-de-compra route', () => {
    renderWithRouter(['/usuario/historial-de-compra']);
    expect(screen.getByText('Historial De Compra')).toBeInTheDocument();
  });

  test('renders EstadisticasPersonales component on /estadisticas-personales route', () => {
    renderWithRouter(['/usuario/estadisticas-personales']);
    expect(screen.getByText('Estadisticas Personales')).toBeInTheDocument();
  });
});