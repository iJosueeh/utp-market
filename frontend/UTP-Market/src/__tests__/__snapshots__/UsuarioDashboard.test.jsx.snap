// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UsuarioDashboard > renderiza el Sidebar y la ruta por defecto (DatosPersonales) 1`] = `
<DocumentFragment>
  <div
    class="usuario-dashboard"
  >
    <div
      data-testid="sidebar"
    >
      Sidebar Closed
    </div>
    <div
      class="main-content"
    >
      <button
        class="sidebar-toggle d-lg-none "
      >
        <i
          class="bi bi-list"
        />
      </button>
      <div
        class="content-wrapper"
      >
        <div>
          Datos Personales
        </div>
      </div>
    </div>
    <style>
      
        .usuario-dashboard {
          display: flex;
        }
        .main-content {
          flex: 1;
          padding: 20px;
          position: relative;
        }
        .sidebar-toggle {
          position: absolute;
          top: 20px;
          left: 20px;
          z-index: 1001;
          background: #222222;
          color: white;
          border: none;
          border-radius: 50%;
          width: 40px;
          height: 40px;
          font-size: 20px;
          display: flex;
          align-items: center;
          justify-content: center;
          padding: 0;
          line-height: 0;
        }
        .sidebar-toggle.hidden {
          display: none;
        }
        @media (max-width: 992px) {
          .content-wrapper {
            padding-top: 50px;
          }
        }
      
    </style>
  </div>
</DocumentFragment>
`;
